pkg/pillar/cipher/handlecipher.go

```diff
@@ -169,18 +155,22 @@ func DecryptCipherBlock(ctx *DecryptCipherContext,
 }
 
 func decryptCipherBlockWithECDH(ctx *DecryptCipherContext,
-	cipherBlock types.CipherBlockStatus) ([]byte, error) {
-	cert, err := getControllerCertEcdhKey(ctx, cipherBlock)
+	cipherContext *types.CipherContext, cipherBlock types.CipherBlockStatus) ([]byte, error) {
+	ecdhKey, err := getControllerCertEcdhKey(ctx, cipherContext.ControllerCertKey())
 	if err != nil {
-		log.Errorf("Could not extract ECDH Certificate Information")
+		log.Errorf("ECDH Certificate Key Information get fail")
 		return []byte{}, err
 	}
-	cipherContext := lookupCipherContext(ctx, cipherBlock.CipherContextID)
-	if cipherContext == nil {
-		errStr := fmt.Sprintf("cipher context %s not found\n",
-			cipherBlock.CipherContextID)
+	eveNodeCert := lookupEveNodeCert(ctx, cipherContext.EveNodeCertKey())
+	if eveNodeCert == nil {
+		log.Errorf("Eve Node Certificate get fail")
+		return []byte{}, err
+	}
+	if eveNodeCert.HasError() {
```

Comment:
When would you have errors on the certificate?

Reply:
For validation/siging errors, if any.

Reply:
But I don't see any code in this PR setting an error. Am I missing it?

Reply:
The controller certificate validation, is pending for object cipher. This is already there for MITM, will add that functionality.